torchao/quantization/subclass.py

```diff
@@ -1176,3 +1150,6 @@ def __torch_dispatch__(cls, func, types, args, kwargs):
         raise NotImplementedError(
             f"LinearActQuantizedTensor dispatch: attempting to run {func}, this is not supported"
         )
+
+to_aqt = AffineQuantizedTensor.from_float
+to_laqt = LinearActQuantizedTensor.from_float
```

Comment:
Why the factory function? Seems like this is only about replacing `F.linear(input, weight)` with `F.linear(some_conversion_function(input), weight)` in a model. Could it just be a class that accepts `some_conversion_function` (in this case `to_aqt`) and that's it? Then we could also reuse it for other dtypes.

Reply:
we need to allow filter_fn that can filter by fqn as well

Reply:
actually not sure what this is referring to, `TensorSubclassQuantizer` already accepts `some_conversion_function`, are you saying we can just use `AffineQuantizedTensor.from_float` instead of `to_aqt`?

Reply:
If someone wants to filter, they can recurse the modules and then only replies instances of `F.linear` they care about.

Reply:
is this talking about the definition of `TensorSubclassQuantizer`? and what is the suggestion? I don't quite understand what is been suggested here

Reply:
For example
```
for name, mod in model.named_modules():
    if "xyz" in name:
        continue
    <apply TensorSubclassQuantizer to mod>
```

Reply:
If we're just talking about filtering modules or such, this seems fairly reasonable. I don't think the Tensor subclass needs to have the filter in it. Our higher level module injection function should be able to give that to users.

Reply:
not sure what you mean, `TensorSubclassQuantizer` is the higher level module injection function I think?

Reply:
I thought it's a building block to help us implement our higher level functions (`apply_dynamic_quant` and such).

Reply:
I'm planning to use `TensorSubclassQuantizer` as a top level API. I think we can implement the current `apply_dynamic_quant` by instantiating `TensorSubclassQuantizer` with a specific set of args

Reply:
Ok, can we revisit that after we've added a few more features (such as SDPA)?

Reply:
yeah sure, do we still want to add `TensorSubclassQuantizer`